### PR TITLE
fix: fix various suggestion edge cases

### PR DESCRIPTION
**Pull request by Sparkier**
fix EOT-56
fix EOT-57

* When reloading suggestions, removes existing ones.
* When suggestion is applied, remove other ones and mark as applied.
* For applied suggestion, remove hover listeners.

### CODE REVIEW

File: src/lib/components/prompt/PromptSuggestions.svelte [hunk 1]
```diff
@@ -49,25 +56,26 @@
...
-						onmouseleave={() => setHoveredSuggestion(null)}
-						onfocus={() => setHoveredSuggestion(suggestion)}
-						onblur={() => setHoveredSuggestion(null)}
-						role="button"
-						tabindex="0"
-					>
-						<PromptSuggestion {prompt} {suggestion} {editPrompt} />
-					</div>
-				{/each}
+		{#if gettingSuggestions}{:else if suggestions === undefined || suggestions.length === 0}
```
Comment (cabreraalex) [inline src/lib/components/prompt/PromptSuggestions.svelte:59]: this looks odd. why not just the if?

File: src/lib/components/prompt/PromptSuggestions.svelte [hunk 2]
```diff
@@ -49,25 +56,26 @@
...
+			No suggestions
+		{:else if suggestionApplied > -1}
```
Comment (cabreraalex) [inline src/lib/components/prompt/PromptSuggestions.svelte:61]: I'm not sure this is the best behavior? Maybe we gray them out so you see you can't apply them? It's confusing when they all disappear I know personally I have to save the current prompt or undo it to see other suggestions but not super clear. We can grey out that apply buttons and add tooltips?

File: src/lib/components/prompt/PromptSuggestions.svelte [hunk 3]
```diff
@@ -49,25 +56,26 @@
 		</button>
 	</div>
 	<div class="flex flex-col gap-4 overflow-auto">
```
Comment (cabreraalex) [inline src/lib/components/prompt/PromptSuggestions.svelte:58]: I'm on the plus plan and getting a paywall every time I try to save

**Review by cabreraalex — CHANGES_REQUESTED** [review]

File: src/lib/components/prompt/PromptSuggestions.svelte (hunk 3 above)
Comment (Sparkier) [reply src/lib/components/prompt/PromptSuggestions.svelte:65]: can't really test this now I'm not on plus and I don't have the correct stripe key

**Comment by cabreraalex** [comment]
Bump now that you have a plus account to check the paywall

**Comment by Sparkier** [comment]
@cabreraalex

**Review by cabreraalex — APPROVED** [review]

**Pull request merged**